resources/assets/lib/chat/conversation-view.tsx

```diff
@@ -80,7 +80,10 @@ export default class ConversationView extends React.Component<Props> implements
       this.didSwitchChannel = false;
     } else {
       snapshot = snapshot ?? blankSnapshot();
-      const prepending = this.firstMessage !== this.currentChannel?.messages[0];
+      const currentChannelFirstMessage = (this.currentChannel != null && this.currentChannel.messages.length > 0)
+        ? this.currentChannel?.messages[0]
+        : null;
+      const prepending = this.firstMessage !== currentChannelFirstMessage;
```

Comment:
should probably add a `firstMessage` getter to Channel that does the check instead.